orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy

```diff
@@ -30,6 +30,8 @@ import retrofit.RetrofitError
 @Component
 @CompileStatic
 class UpsertApplicationTask extends AbstractFront50Task {
+  private static final String VALID_NAME_REGEX = '^[a-zA-Z0-9-]+$'
```

Comment:
This regex is too restrictive ... we have applications with underscores and periods. 

Certain AWS features might not be available depending on names _but_ we haven't prevented those names from being used historically.

Reply:
AWS also doesn't allow `-` in app names either, right?

Reply:
Frigga doesn't allow it, but I _don't think_ AWS particularly cares about dashes for ASGs (at least).